source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -344,7 +344,7 @@ DnsResolverImpl::AddrInfoPendingResolution::AddrInfoPendingResolution(
     DnsResolverImpl& parent, ResolveCb callback, Event::Dispatcher& dispatcher,
     ares_channel channel, const std::string& dns_name, DnsLookupFamily dns_lookup_family)
     : PendingResolution(parent, callback, dispatcher, channel, dns_name),
-      dns_lookup_family_(dns_lookup_family) {
+      dns_lookup_family_(dns_lookup_family), available_interfaces_(availableInterfaces()) {
```

Comment:
Out of curiousity, why pass+store available_interfaces_ instead of simply updating the lookup family?

Reply:
updating the lookup family required adding logic for all possible scenarios between configured lookup family and the available interfaces to then select the new lookup family. It seemed easier to follow to synchronously finish resolution with no address, and a failure than to follow the transition from the configured setting to the new setting based on the interfaces.